images.go

```diff
@@ -219,17 +219,22 @@ func (c *Client) CreateImageUpload(ctx context.Context, opts ImageCreateUploadOp
 
 // UploadImageToURL uploads the given image to the given upload URL.
 func (c *Client) UploadImageToURL(ctx context.Context, uploadURL string, image io.Reader) error {
-	// Linode-specific headers do not need to be sent to this endpoint
-	req := resty.New().SetDebug(c.resty.Debug).R().
-		SetContext(ctx).
-		SetContentLength(true).
-		SetHeader("Content-Type", "application/octet-stream").
-		SetBody(image)
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, uploadURL, image)
+	if err != nil {
+		return err
+	}
 
-	_, err := coupleAPIErrors(req.
-		Put(uploadURL))
+	req.Header.Set("Content-Type", "application/octet-stream")
+	req.ContentLength = -1 // Automatically calculate content length
 
-	return err
+	resp, err := c.httpClient.Do(req)
+
+	_, err = coupleAPIErrors(resp, err)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
I noticed this test fails when manually running the fixtures, still trying to pinpoint why 🤔

Reply:
Interesting, I'll look into this as well. Nice find!

Reply:
As discussed in Slack, this issue turned out to be caused by the S3 REST API not playing nicely with the Transfer-Encoding used by the net/http package when Content-Length isn't explicitly set to something >= 0.

The test was still failing even after hacking together a fix for that on my end, and it turns out the [test client's transport](https://github.com/ezilber-akamai/linodego/blob/TPT-2965-drop-resty/test/integration/integration_suite_test.go#L161-L164) was appending the `Authorization` header to the upload request despite it not using the `Client{}.createRequest(...)` method. One solution might be to simply delete the authorization header before sending the upload request 🙂

Reply:
Thanks for digging into this! Your suggestion to override the user-defined roundtripper seems to have fixed the issue.

Reply:
@ezilber-akamai Thanks for addressing this! Weirdly enough I'm still seeing intermittent errors on the `TestImage_Upload` test, although now they're reporting that the `Content-Length` header is missing. 

It might make sense to use an [io.Seeker](https://stackoverflow.com/a/75459246) to manually discover the buffer size and set the `Content-Length` header accordingly, which will prevent Go from using a chunked transfer encoding 👍

Reply:
This should be fixed now!